api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/common/ActivePatientSnapshotReportBuilder.java

```diff
@@ -213,6 +209,8 @@ protected PatientDataSetDefinition activePatientsDataSetDefinition(String datase
         dsd.addColumn("Current Regimen Line", new ETLCurrentRegLineDataDefinition(), "");
         dsd.addColumn("Baseline CD4", new BaselineCD4CountDataDefinition(), "");
         dsd.addColumn("Date of Baseline CD4 test", new BaselineCD4DateDataDefinition(), "");
+        dsd.addColumn("Latest CD4 Count", new ObsForPersonDataDefinition("Latest CD4 Count", TimeQualifier.LAST, Dictionary.getConcept(Dictionary.CD4_COUNT), null, null), "", new ObsValueNumericConverter(1));
```

Comment:
I propose that we use ETL for this. Pulling these from obs will have negative effects in high volume facilities

Reply:
Ok, noted